From 010c19ba61a3180dab6350aadc578414ad0c0051 Mon Sep 17 00:00:00 2001
From: Conrad Kostecki <conikost@gentoo.org>
Date: Mon, 1 Jan 2024 20:34:19 +0100
Subject: [PATCH 1/2] src/openrct2/core/FileStream.cpp: drop ftello64, fseeko64

The static usage definition of ftello64 and friends is not correct.
While this currently works on glibc, this breaks on musl, as musl
already is LFS aware. The solution is to drop this and add instead
'-D_FILE_OFFSET_BITS=64' to the build system.

Signed-off-by: Conrad Kostecki <conikost@gentoo.org>
---
 src/openrct2/core/FileStream.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/openrct2/core/FileStream.cpp b/src/openrct2/core/FileStream.cpp
index 90a7f7a366cc..6fd7b7b1b42c 100644
--- a/src/openrct2/core/FileStream.cpp
+++ b/src/openrct2/core/FileStream.cpp
@@ -21,11 +21,6 @@
 #    include <io.h>
 #endif
 
-#if defined(__linux__) && !defined(__ANDROID__)
-#    define ftello ftello64
-#    define fseeko fseeko64
-#endif
-
 #ifdef _MSC_VER
 #    define ftello _ftelli64
 #    define fseeko _fseeki64

From 7c638c786527a26a3e83d155ca8ff4d55f01347f Mon Sep 17 00:00:00 2001
From: Conrad Kostecki <conikost@gentoo.org>
Date: Mon, 1 Jan 2024 20:23:24 +0100
Subject: [PATCH 2/2] CMakeLists.txt: add D_FILE_OFFSET_BITS=64

Signed-off-by: Conrad Kostecki <conikost@gentoo.org>
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>
---
 src/openrct2/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/openrct2/CMakeLists.txt b/src/openrct2/CMakeLists.txt
index 367939c9622c..55882a1097d3 100644
--- a/src/openrct2/CMakeLists.txt
+++ b/src/openrct2/CMakeLists.txt
@@ -265,3 +265,7 @@ else ()
     # Dummy target to ease invocation
     add_custom_target(${PROJECT_NAME}-headers-check)
 endif ()
+
+if (UNIX)
+    add_definitions(-D_FILE_OFFSET_BITS=64)
+endif ()
